clients/rust/src/utils.rs

```diff
@@ -1,4 +1,22 @@
+use crate::{errors::MplTokenMetadataError, types::TokenStandard};
+
+/// The offset of the token standard byte in the master edition
+/// from the end of the account data.
+const TOKEN_STANDARD_OFFSET: usize = 1;
+
 /// Removes all null bytes from a string.
 pub fn clean(value: String) -> String {
     value.replace('\0', "")
 }
+
+/// Checks that the `master_edition` is Programmable NFT master edition.
+pub fn assert_edition_is_programmable(edition_data: &[u8]) -> Result<(), MplTokenMetadataError> {
+    if !edition_data.is_empty()
+        && edition_data[edition_data.len() - TOKEN_STANDARD_OFFSET]
```

Comment:
wdyt about something like:
```
edition_data.get(edition_data.len() - TOKEN_STANDARD_OFFSET).ok_or(MplTokenMetadataError::InvalidTokenStandard)?
```

Reply:
I think the check of `edition_data.is_empty()` already prevents any indexing error – even if the length is only `1`, there won't be any invalid index.

Reply:
Unless `TOKEN_STANDARD_OFFSET` ever became something other than 1, which seems unlikely based on the current design.  And even if it did change, someone would have to pass in an invalid edition account, and then still the only fallout would be a harder to debug error.

Still my style advice would be to use `get()` which is 100% impossible to have an issue, but this is fine as well given the likelihood and impact of failure is so minimal.

Reply:
I think I got the best balance now. Instead of checking if the data is empty, I am checking that the data length has to be greater than the offset. This guarantees that the data must include 1 byte for the account discriminator and the length required to index the offset, without needed extra range checks.

Reply:
The `if edition_data.len() > TOKEN_STANDARD_OFFSET` is perfect imo